cylc/flow/task_job_mgr.py

```diff
@@ -306,10 +306,12 @@ def submit_task_jobs(self, workflow, itasks, curve_auth,
 
                     # Get another platform, if task config platform is a group
                     use_next_platform_in_group = False
-                    if itask.tdef.rtconfig['platform']:
+                    bc_mgr = self.task_events_mgr.broadcast_mgr
+                    rtconf = bc_mgr.get_updated_rtconfig(itask)
+                    if rtconf['platform']:
```

Comment:
Looking at this line, I've found another similar bug made possible by this fix.

The only case where`if rtconfig['platform']` evaluates as `False` is if the platform is unset. If it does evaluate `False`, the job will submit with `itask.platform` which is what's defined from the config. With the new broadcast awareness, it is possible to get a new bug out of this when using a broadcast to unset the platform.

Here's an example:

```ini
[runtime]
  [[one]]
    platform = my-platform
```

Then start the workflow and unset the platform:

```console
$ cylc vip --pause
$ cylc broadcast -n one -s 'platform='
$ cylc play
```

The job will attempt to submit to localhost, but with the config of `my-platform`? It fails anyway.

Here's a diff that will reproduce the "out of hosts" scenario without having to mess with SSH config:

```diff
diff --git a/cylc/flow/task_job_mgr.py b/cylc/flow/task_job_mgr.py
index 9eec82970..c0da0e6c0 100644
--- a/cylc/flow/task_job_mgr.py
+++ b/cylc/flow/task_job_mgr.py
@@ -292,13 +292,15 @@ class TaskJobManager:
             for itask in itasks:
                 # If there are any hosts left for this platform which we
                 # have not previously failed to contact with a 255 error.
-                if any(
-                    host not in self.task_remote_mgr.bad_hosts
-                    for host in itask.platform['hosts']
-                ):
-                    platform = itask.platform
-                    out_of_hosts = False
-                    break
+                if False:
+                    pass
+                # if any(
+                #     host not in self.task_remote_mgr.bad_hosts
+                #     for host in itask.platform['hosts']
+                # ):
+                #     platform = itask.platform
+                #     out_of_hosts = False
+                #     break
                 else:
                     # If there are no hosts left for this platform.
                     # See if you can get another platform from the group or
@@ -308,6 +310,8 @@ class TaskJobManager:
                     use_next_platform_in_group = False
                     bc_mgr = self.task_events_mgr.broadcast_mgr
                     rtconf = bc_mgr.get_updated_rtconfig(itask)
+                    LOG.critical(f"#1 {rtconf['platform']}")
+                    LOG.critical(f"#2 {itask.platform['name']}")
                     if rtconf['platform']:
                         try:
                             platform = get_platform(
```

```
INFO - [1/one:waiting] => preparing
CRITICAL - #1 
CRITICAL - #2 localhost
ERROR - [jobs-submit cmd] (remote init)
    [jobs-submit ret_code] 1
CRITICAL - [1/one/01:preparing] submission failed
```

Reply:
I can't see any benefit, other than possibly a minor speed-up from breaking the loop iteration asap. Testing.

Reply:
What it's achieving is not trying to select another platform from a platform group if there are any hosts left on this one.

Reply:
I don't think `if rtconf['platform']:` will achieve that?

Reply:
Red Herring (I was talking about a different if statement).

However, the if statement is required for the case where platform is set by remote/job settings as per #4570 . I think that I've fixed it by passing the whole rtconfig to the get platform routine, which should handle both scenarios.

Reply:
Oh heck, that's a good point. Presumably this logic would also have failed if you attempted to broadcast Cylc-7 style host/batch-system settings too?!

Reply:
Very likely.

Reply:
Resolved - happy with the fix